code/control/LDAPSecurityController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Class SAMLController
+ *
+ * This controller handles serving metadata requests for the IdP, as well as handling
+ * creating new users and logging them into SilverStripe after being authenticated at the IdP.
+ */
+class LDAPSecurityController extends Security {
+
+	/**
+	 * @var array
+	 */
+	private static $allowed_actions = array(
+		'index',
+		'lostpassword',
+		'LostPasswordForm',
+		'ChangePasswordForm'
+	);
+
+	/**
+	 * @param Member $member
+	 * @param $autologinToken
+	 * @return string
+	 */
+	public static function getPasswordResetLink($member, $autologinToken) {
+		$autologinToken = urldecode($autologinToken);
+		$selfControllerClass = __CLASS__;
+		$selfController = new $selfControllerClass();
+		return $selfController->Link('changepassword') . "?m={$member->ID}&t=$autologinToken";
+	}
+
+	/**
+	 * Factory method for the lost password form
+	 *
+	 * @return Form Returns the lost password form
+	 */
+	public function ChangePasswordForm() {
+		return Object::create('LDAPChangePasswordForm', $this, 'ChangePasswordForm');
+	}
+
+	/**
+	 * Factory method for the lost password form
+	 *
+	 * @return Form Returns the lost password form
+	 */
+	public function LostPasswordForm() {
+		$email = new EmailField('Email', _t('Member.EMAIL', 'Email'));
```

Comment:
Logins [handle it gracefully](https://github.com/silverstripe/silverstripe-activedirectory/blob/master/code/authenticators/LDAPAuthenticator.php#L63) though - if the user is not synced yet, a new Member will be created.

I think this is a feature that is useful to have here too, because the typical scenario could be:
- admin person adds a new user to AD
- triggers a password-reset email
- the user tries to reset the password (and fails because the sync is nightly)